torch/csrc/autograd/record_function.cpp

```diff
@@ -4,20 +4,23 @@
 namespace torch { namespace autograd { namespace profiler {
 
 namespace {
-bool has_callbacks = false;
+RecordFunctionState record_function_state = RecordFunctionState::NO_CALLBACKS;
 std::vector<RecordFunctionCallback> start_callbacks;
 std::vector<RecordFunctionCallback> end_callbacks;
+std::vector<bool> needs_inputs_flags({false});
```

Comment:
I'm confused. Why is this a vector again? Deserves a comment at least

Reply:
This is vector because we push/pop callbacks, if we pop a callback we need to know whether the rest of the callbacks need inputs